pkg/cluster/install.go

```diff
@@ -232,64 +255,165 @@ func (m *manager) finishInstallation(ctx context.Context) error {
 		doc.OpenShiftCluster.Properties.Install = nil
 		return nil
 	})
-	return err
+	if err != nil {
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		m.log.Error(err)
+		return api.NewCloudError(
+			http.StatusInternalServerError,
+			api.CloudErrorCodeDeploymentFailed,
+			"", "Failed to finish installation",
+		)
+	}
+	return nil
 }
 
 // initializeKubernetesClients initializes clients which are used
 // once the cluster is up later on in the install process.
 func (m *manager) initializeKubernetesClients(ctx context.Context) error {
 	restConfig, err := restconfig.RestConfig(m.env, m.doc.OpenShiftCluster)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		m.log.Error(err)
+		return api.NewCloudError(
+			http.StatusInternalServerError,
+			api.CloudErrorCodeDeploymentFailed,
+			"", "Failed to initialize the Kubernetes rest config",
```

Comment:
Should I be concerned as a customer?

Reply:
I've removed all of these in favour of reducing the PR's scope.